### ISSUE TITLE
bug: vscode predefined variables not usable

### ISSUE THREAD

**Issue by qizidog** [stevearc/overseer.nvim#181]
### Neovim version (nvim -v)

v0.9.1

### Operating system/version

MacOS 12.6.6 (Apple silicon)

### Describe the bug

I'm not able to use vscode predefined variables to compile with gcc.

The task.json works well in vscode but predefined variables are repalced with '' empty string.

What's more, variables repalcing with '/n' doesn't work well.

### Steps To Reproduce

I can do compile with the minimal.lua below (nvim -u minimal.lua) but it doesn't work well with my LazyVim configuration.

If any infomation is needed, please notify me.

### Expected Behavior

Compile and execute just like what vscode do.

### Minimal example file

`tasks.json`
```json
{
    "version": "2.0.0",
    "tasks": [
        {
            "label": "echo",
            "type": "shell",
            "command": "echo Hello"
        },
        {
            "label": "Compile Current File",
            "type": "shell",
            "command": "gcc",
            "args": [
                "-g",
                "${file}",
                "-O0",
                "-o",
                "${fileBasenameNoExtension}"
            ]
        },
        {
            "label": "Run Current File",
            "command": "./${fileBasenameNoExtension}",
            "dependsOn": ["Compile Current File"]
        },
        {
            "label": "echo predefined variables",
            "type": "shell",
            "command": "echo",
            "args": [
                "userHome: ${userHome}\n",
                "file: ${file}\n",
                "fileBasename: ${fileBasename}\n",
                "fileBasenameNoExtension: ${fileBasenameNoExtension}\n"
            ]
        }
    ]
}
```

c src file to be compiled
```c
#include <stdio.h>
#include <stdlib.h>

int main()
{
  printf("hello, world!\n");
  return EXIT_SUCCESS;
}
```

### Minimal init.lua

```Lua
-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({
    "git",
    "clone",
    "--filter=blob:none",
    "--single-branch",
    "https://github.com/folke/lazy.nvim.git",
    lazypath,
  })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  "folke/tokyonight.nvim",
  { "stevearc/dressing.nvim", config = true },
  {
    "stevearc/overseer.nvim",
    config = function()
      require("overseer").setup({
        -- add your overseer config here
      })
    end,
  },
  -- add any other plugins here
{
    "stevearc/overseer.nvim",
    cmd = {
      "OverseerOpen",
      "OverseerClose",
      "OverseerToggle",
      "OverseerSaveBundle",
      "OverseerLoadBundle",
      "OverseerDeleteBundle",
      "OverseerRunCmd",
      "OverseerRun",
      "OverseerInfo",
      "OverseerBuild",
      "OverseerQuickAction",
      "OverseerTaskAction ",
      "OverseerClearCache",
    },
    opts = {
      task_list = { -- this refers to the window that shows the result
        direction = "bottom",
        default_detail = 1,
        bindings = {
          ["q"] = function()
            vim.cmd("OverseerClose")
          end,
          ["<C-l>"] = false,
          ["<C-h>"] = false,
          ["<C-j>"] = false,
          ["<C-k>"] = false,
          ["L"] = "IncreaseAllDetail",
          ["H"] = "DecreaseAllDetail",
          ["J"] = "ScrollOutputDown",
          ["K"] = "ScrollOutputUp",
        },
      },
      -- Configure the floating window used for task templates that require input
      -- and the floating window used for editing tasks
      form = {
        -- Set any window options here (e.g. winhighlight)
        win_opts = {
          winblend = 0,
        },
      },
      -- Configure the floating window used for confirmation prompts
      confirm = {
        -- Set any window options here (e.g. winhighlight)
        win_opts = {
          winblend = 0,
        },
      },
      -- Configuration for task floating windows
      task_win = {
        -- Set any window options here (e.g. winhighlight)
        win_opts = {
          winblend = 0,
        },
      },
    },
    keys = {
      { "<leader>ct", "<cmd>OverseerRun<cr><cmd>OverseerOpen<cr>", desc = "Run Task" },
      { "<leader>cT", "<cmd>OverseerToggle<cr>", desc = "Toggle Task Panel" },
    },
  }
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

vim.cmd.colorscheme("tokyonight")
-- add anything else here
```

### Additional context

Error executing vim.schedule lua callback: ...im/lazy/overseer.nvim/lua/overseer/task_list/sidebar.lua:315: 'replacement string' item contains newlines
stack traceback:
	[C]: in function 'nvim_buf_set_lines'
	...im/lazy/overseer.nvim/lua/overseer/task_list/sidebar.lua:315: in function 'render'
	.../nvim/lazy/overseer.nvim/lua/overseer/task_list/init.lua:19: in function 'rerender'
	...verseer.nvim/lua/overseer/component/display_duration.lua:34: in function ''
	vim/_editor.lua: in function <vim/_editor.lua:0>

**Comment by qizidog** [comment]
The error msg:

```bash
FAILURE: Compile Current File
gcc -g  -O0 -o 
00:00
out: clang: error: argument to '-o' is missing (expected 1 value)
out: clang: error: no input files
```

telescope.nvim = master(2d92125) latest
dressing.nvim = master(6bde51a) latest

**Comment by qizidog** [comment]
Sorry, I have figured out the key point. It's caused by the keymapping, the focused file changed into the overseer window and thus $(file) is not reachable.

```lua
keys = {
      { "<leader>ct", "<cmd>OverseerRun<cr><cmd>OverseerOpen<cr>", desc = "Run Task" },
}
```

I changed the keymapping into `OverseerOpen!` to solve this.

```lua
keys = {
      { "<leader>ct", "<cmd>OverseerRun<cr><cmd>OverseerOpen!<cr>", desc = "Run Task" },
}
```

However, is there anyway to both solve this problem and change focus to overseer task window?

The problem about ⌘ with `\n` (`Error executing vim.schedule lua callback: ...im/lazy/overseer.nvim/lua/overseer/task_list/sidebar.lua:315: 'replacement string' item contains newlines`) will be reported in another issue.
